game.py

```diff
@@ -67,18 +70,53 @@ def removePlayer(self, player: Player):
     def handleAction(self, player: Player, action: str, row: int, col: int):
         with self.lock:
             # Reject the action if there are not enough players
-            if len(self.players) < MIN_PLAYERS or len(self.players) >= MAX_PLAYERS:
+            if len(self.players) < MIN_PLAYERS or len(self.players) >= MAX_PLAYERS or self.isGameFinished is True:
                 return
             self.isPlaying = True
             box = self.boxes[row][col]
             if action == Action.HOLD and box.canBeHeld(player):
                 self.broadcast(f"{Action.HOLD} {row} {col} {player.id}")
             elif action == Action.CLAIM and box.canBeClaim(player):
-                self.broadcast(f"{Action.CLAIM} {row} {col} {player.id}")
+                winner_index = self.checkWinners()
+                self.broadcast(f"{Action.CLAIM} {row} {col} {player.id} {winner_index}")
+                if winner_index != -1: 
+                    self.isGameFinished = True
             elif action == Action.RELEASE and box.canBeReleased(player):
                 self.broadcast(f"{Action.RELEASE} {row} {col} {player.id}")
 
     # Broadcast a command to all players
-    def broadcast(self, message: str):
+    def broadcast(self, message: str):      
         for player in self.players:
-            player.socket.send(message.encode("ascii"))
+            action, *_ = message.split(" ")
+            new_message = message + f" {player.id}"
+            if action == Action.CLAIM:
+                player.socket.send(new_message.encode("ascii"))
+            else:
+                player.socket.send(message.encode("ascii"))
```

Comment:
There's no need to append the playerId end of the message. The playerId is already assigned to the player when they first connects to the game

Reply:
I removed it. Tks